### ISSUE TITLE
Use backbone 1.2.0

### ISSUE THREAD

**Issue by johannes-photobox** [marionettejs/backbone.marionette#2559]
At the moment when you install backbone and marionette without a version number you will install marionette 2.4.1 and backbone 1.2.0. Since marionette uses 1.1.2, you will have two versions of backbone in use within you application which leads to nasty errors, for example if you want to assign jQuery to backbone `Backbone.$ = require('jquery')` then Marionette can't access jQuery (for Deferred etc.) since it's accessing it's own backbone dependency.

**Comment by rwhitmire** [comment]
+1

**Comment by ahumphreys87** [comment]
Will be fixed with: https://github.com/marionettejs/backbone.marionette/pull/2561

**Comment by JSteunou** [comment]
Would be awesome!

**Comment by paulfalgout** [comment]
@rwhitmire the dependencies were set this way because neither backbone or underscore follow semver, but Marionette does.  So we have to manually updated the dependencies for versions when we know there are no breaking changes.

**Comment by rwhitmire** [comment]
Make sense. I worked around the issue by loading the libraries directly into the browser rather than loading through webpack. Thanks for clarifying!

**Comment by sehmaschine** [comment]
any workaround in the meantime? because the mentioned solution doesn't seem to work (well, at least with my setup) ...

I've installed (via npm):
* backbone 1.1.2
* backbone.marionette 2.4.1
* jquery 2.1.4

For testing purposes, my main.js just contains:
```
var $ = require('jquery');
var Backbone = require('backbone');
var Marionette = require('backbone.marionette');
Backbone.$ = $;
Marionette.$ = Backbone.$;
```

Still, I do see "Uncaught TypeError: Cannot read property 'Deferred' of undefined" referring to `Marionette.Deferred = Backbone.$.Deferred;`.

Any hints/ideas?

**Comment by jridgewell** [comment]
@sehmaschine: You're requiring `Marionette` before you set `Backbone.$`. Instead:

```js
var $ = require('jquery');
var Backbone = require('backbone');
Backbone.$ = $;
var Marionette = require('backbone.marionette');
```

**Comment by sehmaschine** [comment]
@jridgewell that doesn't make a difference (already tried all kinds of variations).

**Comment by jgerigmeyer** [comment]
FWIW, you can workaround this issue using Webpack by specifying an ``alias`` for ``backbone`` and ``underscore``, to force Webpack to use the latest versions only:

package.json:
```js
"backbone": "1.2.0",
"backbone.marionette": "2.4.1",
"underscore": "1.8.3"
```

webpack.config.js
```js
alias: {
  // Force all modules to use versions of backbone and underscore defined
  // in package.json to prevent duplicate dependencies
  'backbone': path.join(__dirname, 'node_modules', 'backbone', 'backbone.js'),
  'underscore': path.join(__dirname, 'node_modules', 'underscore', 'underscore.js')
}
```

**Comment by johannes-photobox** [comment]
@jgerigmeyer but this might break some functionality of Marionette since it doesn't support Backbone 1.2.0 yet. I think the best workaround for now is:

```JavaScript
"dependencies": {
  "backbone": "~1.1.2",
  "backbone.marionette": "^2.4.1",
}
```
;) Until the next update at least...

**Comment by stevewillard** [comment]
Just want to confirm that things seem broken when you use Backbone 1.2.0 (at least, when using Webpack).

```
this.channel = _.result(this, 'channel') || Backbone.Wreqr.radio.channel(this.channelName);
```

It says that it can't read "radio" of undefined -- looks like Wreqr is not being attached to the Backbone object. I can provide more details if it helps anyone.

**Comment by paulfalgout** [comment]
@stevewillard I suspect this is due to `wreqr` also having it's dependencies set to a lower version.  https://github.com/marionettejs/backbone.wreqr/blob/master/package.json#L42

I'm not a webpack user, but it seems like the alias listed above would solve that issue.

**Comment by jgerigmeyer** [comment]
I understand it's not officially supported, but FWIW I've been using Backbone 1.2.0 with Underscore 1.8.3 and Marionette 2.4.1 without trouble so far.

**Comment by stevewillard** [comment]
@paulfalgout Unfortunately, I tried that and it doesn't seem to work.

I think I'm done messing with configuration files for now... I'll just wait for Marionette to bump the Backbone dependency and hopefully things will sort out.

**Comment by realalexhomer** [comment]
+1

I cannot figure out a way currently to use Browserify with Marionette.

I am using Backbone 1.1.2 and Marionette 2.4.1 with the latest version of Browserify to set up a new project and this issue (Cannot read property 'Deferred' of undefined) is essentially stopping me from getting anywhere. I've tried everything suggested in this and other threads I've seen to no avail.

Can anyone think of a good workaround other than loading the files directly into the browser?

**Comment by realalexhomer** [comment]
Whoops, got it working just after I posted that comment. Doing a fresh npm install fixed the issue (after downgrading to Backbone 1.1.2 from 2.0.0).  :neckbeard:

**Comment by culshaw** [comment]
So I had a quick look at https://github.com/marionettejs/marionette-integrations/tree/master/browserify

They are  using a shim to attach jQuery to Backbone, it works in this respect and I've tried it myself with Backbone 1.1.2.

check the package.json for the build.

**Comment by ahumphreys87** [comment]
Backbone dependency has been bumped to 1.2.1 and will be released in v2.4.2

**Comment by stevewillard** [comment]
Out of curiosity, is there an ETA on when v2.4.2 will be released?

**Comment by ahumphreys87** [comment]
hopefully in the next couple of days - we are actually going through as I
type :smile:

**Comment by quentindemetz** [comment]
I think I'm having the same issue again, with Marionette loading Backbone@1.2.1 and Wreqr loading Backbone@1.2.2, which means accessing Backbone.Wreqr within Marionette code does not call the version of Backbone which has Wreqr attached to it...

**Comment by ahumphreys87** [comment]
Which version of backbone are you using in your app?

We are waiting on BB 1.2.3 before releasing Mn 2.4.3 which will include the
dependency bump

**Comment by quentindemetz** [comment]
I was using 1.2.2, rolling back to 1.2.1 fixed the issue. :+1:

**Comment by newtonianb** [comment]
@stevewillard @quentindemetz 
We are having the exact same issue! how did you go about solving this? We are trying a bunch of different versions but still no luck.
http://stackoverflow.com/questions/35802424/backbone-marionette-and-webpack-uncaught-typeerror-cannot-read-property-radi

**Comment by libahipi** [comment]
@newtonianb 
Had the same issue today with browserify/backbone/marionette setup after Backbone 1.3.1 update. Reverting Backbone to 1.2.3 fixed it.
`"backbone": "1.2.3",
    "backbone.marionette": "^2.4.4",`

**Comment by paulfalgout** [comment]
XRef: https://github.com/jashkenas/backbone/issues/3975#issuecomment-192386575

**Comment by newtonianb** [comment]
@libahipi @paulfalgout thanks very much

**Comment by salex772** [comment]
For me the only way to solve it is backbone 1.2.0 and marionette 2.4.3 and lodash < ^3 (lodash may be not related to this issue and is only my configuration additional issue)

**Comment by Growiel** [comment]
I went with @jgerigmeyer 's suggestion after losing a lot of time. Seems to be working fine, I can use latest Backbone and Marionette.

For tests sake, removing completely Backbone from package.json and letting Marionette use whatever it wanted was also working, but I like to have new version of stuff whenever possible, and nothing breaking has been released to Backbone as far as I could tell.

**Comment by scott-w** [comment]
@salex772 My understanding is that lodash isn't fully supported in Backbone and, thus, not supported in Marionette 2.x.

@Growiel in current versions of Backbone+Marionette, the following guide should work:
https://marionette.gitbooks.io/marionette-guides/content/en/getting_started/installing_marionette.html

**Comment by salex772** [comment]
I used clean install and still cannot get it work. I run into marionettejs/backbone.marionette#2559 issue
I use npm 3.6.0 and node 5.6.0. But on another computer I have npm 3.5.2 and node 4.2.3 and marionette 2.4.4 works with backbone 1.2.3 without that error.

**Comment by paulfalgout** [comment]
@salex772 until marionette lists backbone 1.3.2 as a dependency, there should be no expectation that this would work.

**Comment by salex772** [comment]
Now I cleaned out packages.json and now it works with Bb 1.2.3

**Comment by bfdes** [comment]
If anyone else is looking at this now: in my case a combination of backbone.marionette@2.4.7 and backbone@1.3.3 seems to work without needing an alias for backbone.
